EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
+                _localizedStringFinder.GetString(EOResourceID.GUILD_REGISTER_GUILD),
+                _localizedStringFinder.GetString(EOResourceID.GUILD_ENTER_YOUR_GUILD_DETAILS),
+                TextMultiInputDialog.DialogSize.Three,
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_TAG), MaxChars = 3 },
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_NAME), MaxChars = 24 },
+                new TextMultiInputDialog.InputInfo { Label = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_DESCRIPTION), MaxChars = 240 }
+            );
+
+            dlgRegister.DialogClosing += (_, e) =>
+            {
+                if (e.Result == XNADialogResult.OK)
+                {
+                    var guildTag = dlgRegister.Responses[0];
+                    var guildName = dlgRegister.Responses[1];
+                    var guildDescription = dlgRegister.Responses[2];
+
+                    if (!_characterInventoryProvider.ItemInventory.Any(x => x.ItemID == 1 && x.Amount >= 50_000))
+                    {
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.WARNING_YOU_HAVE_NOT_ENOUGH, $" {_eifFileProvider.EIFFile[1].Name}");
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (guildTag.Length == 0)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_TAG_FIELD_EMPTY);
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (guildName.Length == 0)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_NAME_FIELD_EMPTY);
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (guildTag.Length == 1)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_TAG_TOO_SHORT);
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (guildName.Length <= 3)
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_NAME_TOO_SHORT);
+                        dlg.Show();
+                        return;
+                    }
+
+                    if (char.ToLower(guildTag[0]) != char.ToLower(guildName[0]))
+                    {
+                        e.Cancel = true;
+                        var dlg = _messageBoxFactory.CreateMessageBox(DialogResourceID.GUILD_CREATE_NAME_FIELD_EMPTY);
```

Comment:
Is there no better error message for this condition (guild tag first character must match guild name first character)?

Reply:
Oops forgot to use it lol `DialogResourceID.GUILD_TAG_NAME_LETTER_MUST_MATCH`